src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/UrlRewriteRuleBuilder.cs

```diff
@@ -19,6 +19,7 @@ public class UrlRewriteRuleBuilder
 
         private UrlMatch _initialMatch;
         private IList<Condition> _conditions;
+        private readonly IList<ServerVariable> _serverVariables = new List<ServerVariable>();
```

Comment:
Based on previous comment with supporting the Replace tag, I'd make this a dict from name to pattern.